src/prefect/agent.py

```diff
@@ -37,7 +37,7 @@
 from prefect.infrastructure import Infrastructure, InfrastructureResult, Process
 from prefect.logging import get_logger
 from prefect.settings import PREFECT_AGENT_PREFETCH_SECONDS
-from prefect.states import Crashed, Pending, StateType, exception_to_failed_state
```

Comment:
If there are no longer code paths that transition to a failed status, then what about all these?

<img width="469" alt="Screen Shot 2023-06-18 at 9 32 27 AM" src="https://github.com/PrefectHQ/prefect/assets/878798/a68edd63-f9c2-4e06-83d3-35853567a381">

PS. I'm not familiar with the prefect codebase, just grabbed a few random tagged with `good first issue`, so review with extreme caution :)

Reply:
If I've understood correctly, failed state is now reserved for failed validation (and maybe timeouts)?

For example, like this:

```
    state = Pending()
    if flow.should_validate_parameters:
        try:
            parameters = flow.validate_parameters(parameters)
        except Exception:
            state = await exception_to_failed_state(
                message="Validation of flow parameters failed with error:"
            )
```

But are we confident that all of the above `await exception_to_failed_state` statements should remain as-is, or should some now become `exception_to_crashed_state`?

(totally new to this codebase, so apologies if this is obvious to everyone else)